src/sql/parts/notebook/cellViews/codeActions.ts

```diff
@@ -106,8 +102,30 @@ export class RunCellAction extends ToggleableAction {
 				this._executionChangedDisposable.dispose();
 			}
 			this._context = context;
-			this.toggle(context.cell.isRunning);
-			this._executionChangedDisposable = this._context.cell.onExecutionStateChange(this._handleExecutionStateChange, this);
+			this.updateStateAndExecutionCount(context.cell.executionState);
+			this._executionChangedDisposable = this._context.cell.onExecutionStateChange((state) => {
+				this.updateStateAndExecutionCount(state);
+			});
+		}
+	}
+
+	private updateStateAndExecutionCount(state: CellExecutionState) {
+		let label = emptyExecutionCountLabel;
+		let className = '';
+		if (!types.isUndefinedOrNull(this._context.cell.executionCount)) {
+			label = `[${this._context.cell.executionCount}]`;
+			// Heuristic to try and align correctly independent of execution count length. Moving left margin
```

Comment:
I think this is ok for now, but maybe create a task to try to clean this up in the future?

Reply:
I'm going to call an audible and say "let's create a task for this when we need it". I suspect any task will be hanging around for a long time. Let's let organic requirements drive this instead (e.g. adding another button with similar issues). Sound OK?